server/util/flagutil/types/types.go

```diff
@@ -258,16 +328,47 @@ type DeprecatedFlag[T any] struct {
 //   "All of our foos were destroyed in a fire, please specify a bar instead.",
 // )
 func DeprecatedVar[T any](value flag.Value, name string, usage, migrationPlan string) *T {
-	common.DefaultFlagSet.Var(&DeprecatedFlag[T]{value, name, migrationPlan}, name, usage+" **DEPRECATED** "+migrationPlan)
 	converted, err := common.ConvertFlagValue(value)
 	if err != nil {
 		log.Fatalf("Error creating deprecated flag %s: %v", name, err)
 	}
-	return converted.(*T)
+	common.DefaultFlagSet.Var(
+		&DeprecatedFlag[T]{
+			Value:         value,
+			name:          name,
+			MigrationPlan: migrationPlan,
+		},
+		name,
+		usage+" **DEPRECATED** "+migrationPlan,
+	)
+	c, ok := converted.(*T)
+	if !ok {
+		log.Fatalf("Error creating deprecated flag %s: could not coerce flag of type %T to type %T.", name, converted, (*T)(nil))
+	}
+	return c
+}
+
+// Deprecate deprecates an existing flag by name; generally this should be
+// called in an init func. While simpler to use than DeprecatedVar, it does
+// decouple the flag declaration from the flag deprecation.
+func Deprecate[T any](name, migrationPlan string) {
+	flg := common.DefaultFlagSet.Lookup(name)
+	converted, err := common.ConvertFlagValue(flg.Value)
+	if err != nil {
+		log.Fatalf("Error creating deprecated flag %s: %v", name, err)
+	} else if _, ok := converted.(*T); !ok {
+		log.Fatalf("Error creating deprecated flag %s: could not coerce flag of type %T to type %T.", name, converted, (*T)(nil))
+	}
+	flg.Value = &DeprecatedFlag[T]{
+		Value:         flg.Value,
+		name:          flg.Name,
+		MigrationPlan: migrationPlan,
+	}
+	flg.Usage = flg.Usage + " **DEPRECATED** " + migrationPlan
 }
```

Comment:
nit: are there any cases where `Deprecate` would be used, where `Deprecated` or `DeprecatedVar` could not? wondering if we could make do with just `Deprecated` and `DeprecatedVar`, so that things are more consistent

Reply:
`Deprecate` lets us reach in and deprecate any flag, including flags from other libraries, if we ever start using libs that define their own flags, which could be quite useful. It also makes the definitions in `autoflags` much simpler. I think what I should maybe do, actually, is define `Deprecated` and `DeprecatedVar` in terms of `Deprecate` to remove code duplication.

Reply:
Oh, wait, there is no `Deprecated`, because I couldn't make that without `autoflags`, that's right. Well, I'll define `DeprecatedVar` in terms of `Deprecate`, at least.